src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -168,17 +163,14 @@ private void handleTick() {
   }
 
   private void maybeRolloverWriterForDay() throws IOException {
-    if (writer == null || !logger.getNow().equals(writerDate)) {
+    if (writer == null || recordsWriterFactory.shouldRollover()) {
       if (writer != null) {
-        // Day changes over case, reset the logFileCount.
-        logFileCount = 0;
         IOUtils.closeQuietly(writer);
         writer = null;
       }
-      // increment log file count, if creating a new writer.
-      ++logFileCount;
-      writer = logger.createWriter(constructFileName());
-      writerDate = DatePartitionedRecordsWriterFactory.getDateFromDir(writer.getPath().getParent().getName());
+
+      writer = recordsWriterFactory.createWriter();
+      recordsWriterFactory.maybeUpdateRolloverTime();
```

Comment:
Would it simplify things if `maybeUpdateRolloverTime` was called from within `createWriter`? Going further: Maybe you could move `writer` to the inside of `recordsWriterFactory` and change the API to:

```
Writer getOrCreateWriter(); // handles the rollover internally
void close(Writer writer);
```